### PR TITLE
Enables parser to build and return an AST, using the `Tree` type.

### LINKED ISSUE #4: V1 Markup Parser

### PR DESCRIPTION
**Pull request by tylercecil**
Depends on #13 and #12 . My apologies for the diff. I didn't realize github doesn't allow for comparison onto synthetic branches. Will chain, next time.

Simply takes the logic laid out in #12 , and captures the relevant tokens, storing their information into an AST. Theoretically, this PR constitutes a fully functioning parser for chomTree. Closes #4 .

### CODE REVIEW

**Comment by tylercecil** [comment]
I made a pretty big refactor to #13 , which I'll merge in here tomorrow. That change will allow me to write up the tests.

**Comment by tylercecil** [comment]
The history on this branch is a nightmare... my apologies...

File: src/parser.ts [hunk 1]
```diff
@@ -53,71 +55,94 @@ function parse(input: string) {
...
   }
 
   function nodeType() {
-    expect('WORD');
+    const nt = { name: '', sub: '', sup: '' };
+    nt.name = expect('WORD').value as string;
 
     if (accept('_')) {
-      expect('WORD');
+      nt.sub = expect('WORD').value as string;
```
Comment (duncgibbs) [inline src/parser.ts:98]: I think if you type the expect function return, you won't have to do type assertion here. But I might be wrong.

Comment (tylercecil) [reply src/parser.ts:98]: Actually, now that I'm actually thinking what I MEANT was `nt.sub = epxect('WORD')!.value`. The problem is value is optional.
I am actually going to leave it as is, though. I've had a change of heart about the tokens, and am hoping to make a big update to the token types once a stable V1 is finished.

Comment (tylercecil) [reply src/parser.ts:98]: Also, I'm too lazy to merge everything up through these other PRs :p

**Pull request merged**